### DISCUSSION TITLE
Home Assistant MQTT sensors

### DISCUSSION THREAD

**Discussion by DrSpaldo** [blakeblackshear/frigate#5185] [Ask A Question]
Has anyone made any good Home Assistant MQTT sensors for info that comes through? 

For example, I would like to keep an eye on how much the shm & tmpfs are using. I have had a look at the output of the mqtt frigate/stats. However, the output is one massive chunk of data, so much data:

`{"frontdoor": {"camera_fps": 5.0, "process_fps": 5.0, "skipped_fps": 0.0, "detection_fps": 5.9, "pid": 558, "capture_pid": 568, "ffmpeg_pid": 575}, "frontdriveway": {"camera_fps": 5.0, "process_fps": 5.1, "skipped_fps": 0.0, "detection_fps": 30.8, "pid": 559, "capture_pid": 569, "ffmpeg_pid": 577}, "garage": {"camera_fps": 5.0, "process_fps": 5.0, "skipped_fps": 0.0, "detection_fps": 0.0, "pid": 560, "capture_pid": 576, "ffmpeg_pid": 586}, "backdoor": {"camera_fps": 5.0, "process_fps": 5.1, "skipped_fps": 0.0, "detection_fps": 4.8, "pid": 561, "capture_pid": 578, "ffmpeg_pid": 588}, "sidegarage": {"camera_fps": 5.0, "process_fps": 5.0, "skipped_fps": 0.0, "detection_fps": 0.0, "pid": 563, "capture_pid": 587, "ffmpeg_pid": 593}, "frontroad": {"camera_fps": 5.0, "process_fps": 5.0, "skipped_fps": 0.0, "detection_fps": 9.6, "pid": 565, "capture_pid": 594, "ffmpeg_pid": 601}, "backyard": {"camera_fps": 5.1, "process_fps": 5.2, "skipped_fps": 0.0, "detection_fps": 10.3, "pid": 566, "capture_pid": 598, "ffmpeg_pid": 615}, "detectors": {"coral1": {"inference_speed": 6.46, "detection_start": 0.0, "pid": 549}, "coral2": {"inference_speed": 6.49, "detection_start": 0.0, "pid": 551}}, "detection_fps": 61.4, "gpu_usages": {"NVIDIA GeForce GTX 1070": {"gpu": "5 %", "mem": "28.1 %"}}, "cpu_usages": {"top": {"cpu": "users,", "mem": "load"}, "Tasks:": {"cpu": "stopped,", "mem": "0"}, "%Cpu(s):": {"cpu": "id,", "mem": "3.1"}, "MiB": {"cpu": "36732.8", "mem": "avail"}, "PID": {"cpu": "%CPU", "mem": "%MEM"}, "559": {"cpu": "70.7", "mem": "0.3"}, "565": {"cpu": "18.3", "mem": "0.3"}, "577": {"cpu": "18.3", "mem": "0.6"}, "558": {"cpu": "16.3", "mem": "0.3"}, "586": {"cpu": "11.0", "mem": "0.4"}, "601": {"cpu": "8.7", "mem": "0.5"}, "553": {"cpu": "3.7", "mem": "0.2"}, "575": {"cpu": "8.3", "mem": "0.5"}, "594": {"cpu": "4.7", "mem": "0.2"}, "598": {"cpu": "4.0", "mem": "0.2"}, "615": {"cpu": "8.3", "mem": "0.6"}, "551": {"cpu": "4.0", "mem": "0.3"}, "560": {"cpu": "0.3", "mem": "0.2"}, "568": {"cpu": "5.3", "mem": "0.2"}, "576": {"cpu": "4.0", "mem": "0.2"}, "587": {"cpu": "4.7", "mem": "0.2"}, "588": {"cpu": "8.7", "mem": "0.5"}, "1": {"cpu": "0.0", "mem": "0.0"}, "15": {"cpu": "0.0", "mem": "0.0"}, "16": {"cpu": "0.0", "mem": "0.0"}, "24": {"cpu": "0.0", "mem": "0.0"}, "25": {"cpu": "0.0", "mem": "0.0"}, "33": {"cpu": "0.0", "mem": "0.0"}, "72": {"cpu": "0.0", "mem": "0.0"}, "74": {"cpu": "0.0", "mem": "0.0"}, "75": {"cpu": "0.0", "mem": "0.0"}, "76": {"cpu": "0.0", "mem": "0.0"}, "77": {"cpu": "0.0", "mem": "0.0"}, "78": {"cpu": "0.0", "mem": "0.0"}, "79": {"cpu": "0.0", "mem": "0.0"}, "81": {"cpu": "53.7", "mem": "1.4"}, "82": {"cpu": "0.0", "mem": "0.0"}, "86": {"cpu": "0.0", "mem": "0.0"}, "87": {"cpu": "0.0", "mem": "0.0"}, "90": {"cpu": "0.0", "mem": "0.0"}, "124": {"cpu": "0.0", "mem": "0.0"}, "125": {"cpu": "0.0", "mem": "0.0"}, "126": {"cpu": "0.0", "mem": "0.0"}, "127": {"cpu": "0.0", "mem": "0.0"}, "133": {"cpu": "0.0", "mem": "0.0"}, "157": {"cpu": "0.0", "mem": "0.0"}, "194": {"cpu": "0.0", "mem": "0.0"}, "240": {"cpu": "0.0", "mem": "0.0"}, "275": {"cpu": "0.0", "mem": "0.0"}, "314": {"cpu": "0.0", "mem": "0.0"}, "353": {"cpu": "0.0", "mem": "0.0"}, "397": {"cpu": "0.0", "mem": "0.0"}, "542": {"cpu": "0.0", "mem": "0.1"}, "548": {"cpu": "0.3", "mem": "0.0"}, "549": {"cpu": "4.0", "mem": "0.2"}, "557": {"cpu": "0.0", "mem": "0.0"}, "561": {"cpu": "7.7", "mem": "0.2"}, "562": {"cpu": "0.0", "mem": "0.0"}, "563": {"cpu": "0.7", "mem": "0.2"}, "564": {"cpu": "0.0", "mem": "0.0"}, "566": {"cpu": "13.3", "mem": "0.3"}, "567": {"cpu": "0.0", "mem": "0.0"}, "569": {"cpu": "7.3", "mem": "0.2"}, "574": {"cpu": "0.0", "mem": "0.0"}, "578": {"cpu": "5.3", "mem": "0.2"}, "584": {"cpu": "0.0", "mem": "0.0"}, "591": {"cpu": "0.0", "mem": "0.0"}, "593": {"cpu": "8.3", "mem": "0.6"}, "597": {"cpu": "0.0", "mem": "0.0"}, "1349": {"cpu": "2.0", "mem": "0.0"}, "4865": {"cpu": "0.0", "mem": "0.0"}, "4876": {"cpu": "0.3", "mem": "0.0"}, "4882": {"cpu": "4.3", "mem": "0.0"}}, "service": {"uptime": 1347, "version": "0.12.0-cf2466c", "latest_version": "0.11.1", "storage": {"/media/frigate/recordings": {"total": 36476413.1, "used": 26337076.2, "free": 10139134.0, "mount_type": "fuse.shfs"}, "/media/frigate/clips": {"total": 36476413.1, "used": 26337076.2, "free": 10139134.0, "mount_type": "fuse.shfs"}, "/tmp/cache": {"total": 1000.0, "used": 31.5, "free": 968.5, "mount_type": "tmpfs"}, "/dev/shm": {"total": 536.9, "used": 59.3, "free": 477.5, "mount_type": "tmpfs"}}, "temperatures": {"apex_0": 64.55, "apex_1": 62.55}}}`

**Comment by NickM-27** [comment]
Some of these storage bits will be added to sensors in frigate integration in the future.

**Reply by Red-ington** [reply]
@DrSpaldo very good question. looks like a great addon.
